scaleway/resource_rdb_privilege_test.go

```diff
@@ -0,0 +1,170 @@
+						user_name     = scaleway_rdb_user.foo.name
+						database_name = scaleway_rdb_database.db.name
+						permission    = "all"
+					}`, instanceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRdbPrivilegeExists(tt, "scaleway_rdb_instance.instance", "scaleway_rdb_database.db", "scaleway_rdb_user.foo"),
+					resource.TestCheckResourceAttr("scaleway_rdb_privilege.priv", "permission", "all"),
+				),
+			},
+			{
+				Config: fmt.Sprintf(`
+					resource "scaleway_rdb_instance" "instance" {
+						name = "%s"
+						node_type = "db-dev-s"
+						engine = "PostgreSQL-12"
+						is_ha_cluster = false
+						tags = [ "terraform-test", "scaleway_rdb_user", "minimal" ]
+					}
+
+					resource "scaleway_rdb_database" "db" {
+						instance_id = scaleway_rdb_instance.instance.id
+						name = "foo"
+					}
+
+					resource "scaleway_rdb_user" "foo" {
+						instance_id = scaleway_rdb_instance.instance.id
+						name = "foo"
+						password = "R34lP4sSw#Rd"
+					}
+
+					resource "scaleway_rdb_privilege" "priv" {
+						instance_id   = scaleway_rdb_instance.instance.id
+						user_name     = scaleway_rdb_user.foo.name
+						database_name = scaleway_rdb_database.db.name
+						permission    = "none"
+					}`, instanceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRdbPrivilegeExists(tt, "scaleway_rdb_instance.instance", "scaleway_rdb_database.db", "scaleway_rdb_user.foo"),
+					resource.TestCheckResourceAttr("scaleway_rdb_privilege.priv", "permission", "none"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckRdbPrivilegeExists(tt *TestTools, instance string, database string, user string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+		instanceResource, ok := state.RootModule().Resources[instance]
+		if !ok {
+			return fmt.Errorf("resource not found: %s", instance)
+		}
+
+		databaseResource, ok := state.RootModule().Resources[database]
+		if !ok {
+			return fmt.Errorf("resource database not found: %s", database)
+		}
+
+		userResource, ok := state.RootModule().Resources[user]
+		if !ok {
+			return fmt.Errorf("resource not found: %s", user)
+		}
+
+		rdbAPI, region, _, err := rdbAPIWithRegionAndID(tt.Meta, instanceResource.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		_, instanceID, databaseName, err := resourceScalewayRdbDatabaseParseID(databaseResource.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		_, userName, err := resourceScalewayRdbUserParseID(userResource.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		databases, err := rdbAPI.ListPrivileges(&rdb.ListPrivilegesRequest{
+			Region:       region,
+			InstanceID:   instanceID,
+			DatabaseName: &databaseName,
+			UserName:     &userName,
+		})
+		if err != nil {
+			return err
+		}
+
+		if len(databases.Privileges) != 1 {
+			return fmt.Errorf("no privilege found")
+		}
+
+		return nil
+	}
+}
+
+func TestValidationPrivilegePermissionWithWrongPermissionReturnError(t *testing.T) {
```

Comment:
Why are those tests required if you use validation defined in the schema?

Reply:
Unit testing :)

But following https://github.com/scaleway/terraform-provider-scaleway/pull/844/commits/046230f1cc4255ae7fdb23fe3e0d700fac63cee9 those tests are removed